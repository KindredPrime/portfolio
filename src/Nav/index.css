.Nav {
  margin-right: 50px;
  padding-left: 30px;
  padding-right: 30px;
  height: 75px;
  background-color: black;
  position: fixed;
  top: 0px;
  left: 0px;
  width: 100%;
  z-index: 1;
  border-bottom: 1px solid #A1A1A1;
}

.Nav.expanded {
  height: 100px;
}

.Nav__home, .Nav__menu {
  position: absolute;
}

.Nav__home {
  top: 20px;
  left: 30px;
}

.Nav__menu {
  right: 30px;
  top: 15px;
  width: 25px;
  height: 25px;
  padding: 0px;
  margin: 0px;
  background-color: transparent;
  color: #00FFFF;
  border: none;
}

.Nav * {
  color: #00FFFF;
}

.Nav__menu-links {
  position: relative;
  top: 57px;
  display: none;
}

.Nav__menu-links.expanded {
  display: flex;
  align-items: center;
  justify-content: flex-end;
}

.Nav__menu-links * {
  display: inline-block;
}

.Nav__menu-links img:hover,
.email-image-link:focus,
.linkedin-link:focus {
  filter: drop-shadow(1px -2px 5px #00FFFF);
}

img:focus {
  transform: scale(2);
}

.email-image-link, .linkedin-link, .github-link {
  margin-left: 7%;
}

.email-icon, .linkedin-icon {
  width: 30px;
  height: 25px;
}

.email-text-link {
  display: none;
}

.github-logo {
  width: 70px;
  height: 30px;
}

/*
  Tablet devices
*/
@media screen and (min-width: 690px) {
  .Nav {
    padding-left: 10%;
    padding-right: 10%;
    height: 100px;
  }

  .Nav__home {
    left: 10%;
  }

  .Nav__menu {
    display: none;
  }

  .Nav__menu-links {
    display: flex;
    align-items: center;
    justify-content: flex-end;
  }

  .email-image-link {
    display: none;
  }

  .email-text-link {
    display: inline-block;
  }

  .email-text-link, .linkedin-link, .github-link {
    margin-left: 4%;
  }
}

/*
  Desktop/laptop computers
*/
@media screen and (min-width: 1000px) {
  .Nav {
    padding-left: 12%;
    padding-right: 12%;
  }

  .Nav__home {
    left: 12%;
  }
}