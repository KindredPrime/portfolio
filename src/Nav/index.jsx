import React from 'react';
import { NavLink } from 'react-router-dom';
import emailIcon from '../images/email-icon.png';
import linkedInIcon from '../images/linkedin-logo.png';
import './index.css';

class Nav extends React.Component {
  state = {
    expanded: false
  };

  render() {
    const { expanded } = this.state;

    const navClass = expanded
      ? "Nav expanded"
      : "Nav";

    const menuLinksClass = expanded
      ? "Nav__menu-links expanded"
      : "Nav__menu-links";

    return (
      <nav className={navClass}>
        <NavLink className="Nav__home" to="/">
          Ken Wright
          <span className="visually-hidden">Change layout to homepage</span>
        </NavLink>

        <button
          className="Nav__menu"
          onClick={() => this.setState({ expanded: !expanded })}
        >
          â˜°
          <span className="visually-hidden">Show/Hide menu links</span>
        </button>

        <div className={menuLinksClass}>
          <NavLink to="/projects">Projects</NavLink>

          <a
            href="mailto:kenneth.gordon.wright@gmail.com"
            target="_blank"
            rel="noreferrer"
            className="email-image-link"
          >
            <img src={emailIcon} alt="Email" className="email-icon" />
          </a>

          <a
            href="mailto:kenneth.gordon.wright@gmail.com"
            target="_blank"
            rel="noreferrer"
            className="email-text-link"
          >
            kenneth.gordon.wright@gmail.com
          </a>

          <a
            href="https://www.linkedin.com/in/ken-wright-52554556/"
            className="linkedin-link"
            target="_blank"
            rel="noreferrer"
          >
            <img src={linkedInIcon} alt="LinkedIn" className="linkedin-icon" />
          </a>

          <a
            href="https://github.com/KindredPrime"
            className="github-link"
            target="_blank"
            rel="noreferrer"
          >
            GitHub
          </a>
        </div>

      </nav>
    );
  }
}

export default Nav;