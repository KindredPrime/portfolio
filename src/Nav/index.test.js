import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import Nav from './';

describe('Nav Component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <BrowserRouter>
        <Nav />
      </BrowserRouter>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it('displays the menu links when the menu button is clicked', () => {
    render(
      <BrowserRouter>
        <Nav />
      </BrowserRouter>
    );

    userEvent.click(screen.getByText('â˜°'));

    expect(document.body).toMatchSnapshot();
  });
});