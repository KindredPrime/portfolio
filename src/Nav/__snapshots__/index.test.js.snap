// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav Component displays the menu links when the menu button is clicked 1`] = `
<body>
  <div>
    <nav
      class="Nav expanded"
    >
      <a
        aria-current="page"
        class="Nav__home active"
        href="/"
      >
        Ken Wright
        <span
          class="visually-hidden"
        >
          Change layout to homepage
        </span>
      </a>
      <button
        class="Nav__menu"
      >
        â˜°
        <span
          class="visually-hidden"
        >
          Show/Hide menu links
        </span>
      </button>
      <div
        class="Nav__menu-links expanded"
      >
        <a
          href="/projects"
        >
          Projects
        </a>
        <a
          class="email-image-link"
          href="mailto:kenneth.gordon.wright@gmail.com"
          rel="noreferrer"
          target="_blank"
        >
          <img
            alt="Email"
            class="email-icon"
            src="email-icon.png"
          />
        </a>
        <a
          class="email-text-link"
          href="mailto:kenneth.gordon.wright@gmail.com"
          rel="noreferrer"
          target="_blank"
        >
          kenneth.gordon.wright@gmail.com
        </a>
        <a
          class="linkedin-link"
          href="https://www.linkedin.com/in/ken-wright-52554556/"
          rel="noreferrer"
          target="_blank"
        >
          <img
            alt="LinkedIn"
            class="linkedin-icon"
            src="linkedin-logo.png"
          />
        </a>
        <a
          class="github-link"
          href="https://github.com/KindredPrime"
          rel="noreferrer"
          target="_blank"
        >
          GitHub
        </a>
      </div>
    </nav>
  </div>
</body>
`;
